src/aws-sdk.ts

```diff
@@ -0,0 +1,98 @@
+import type { Service as AWSService } from "aws-sdk";
+// eslint-disable-next-line import/no-extraneous-dependencies
+import type * as AWS from "aws-sdk";
+import { AnyFunction, OverloadUnion } from "./util";
+
+type AWSServiceClass = { new (): AWSService };
+
+/**
+ * First we have to extract the names of all Services in the v2 AWS namespace
+ *
+ * @returns "AccessAnalyzer" | "Account" | ... | "XRay"
+ */
+export type ServiceKeys = Exclude<
+  {
+    [K in keyof typeof AWS]: typeof AWS[K] extends AWSServiceClass ? K : never;
+  }[keyof typeof AWS],
+  "Service"
+>;
+
+export type SDK = {
+  [serviceName in ServiceKeys]: typeof AWS[serviceName] extends new (
+    ...args: any[]
+  ) => infer Client
+    ? {
+        [methodName in keyof Client]: SdkMethod<Client[methodName]>;
+      }
+    : never;
+};
+
+export interface SdkCallOptions {
+  /**
+   * The resources for the IAM statement that will be added to the state machine
+   * role's policy to allow the state machine to make the API call.
+   *
+   * Use `["*"]` to grant access to all resources (discouraged).
+   *
+   * @example ["arn:aws:s3:::my_bucket"]
+   * @see https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_resource.html
+   */
+  iamResources: string[];
```

Comment:
I wonder if we should re-use the `PolicyDocument` type from the CDK? Or `PolicyStatement`?

Reply:
The [PolicyStatement](https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-iam/lib/policy-statement.ts#L43) would be the correct class here. It would be passed to the call and we could check [`actions`](https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-iam/lib/policy-statement.ts#L539) field and add the default if not present 👍

Reply:
@sam-goodwin would you like the following syntax better:

```ts
$AWS.SDK.CloudWatch.describeAlarms({
  AlarmNames: [alarm.alarmName]
}, new iam.PolicyStatement({resources: ["*"]}));
```